libs/ui/src/unlock_machine_screen.tsx

```diff
@@ -73,7 +73,7 @@ export function UnlockMachineScreen({
           auth.checkPasscode(passcode);
         } else {
           assert(checkPinFromProps !== undefined);
-          await checkPinFromProps(passcode);
+          checkPinFromProps(passcode);
```

Comment:
Just making sure we're ok with not `await`ing here before resetting the passcode to `''`

Reply:
Yes good q - I wasn't able to detect a noticeable difference. Even when we had the await, the passcode clearing was visible before the transition to the remove card screen (because auth status doesn't update until the next re-fetch)

Side note: This is another interesting downside to our current approach to mutations - losing the ability to `await` at code points outside of the mutation `onSuccess` callback

Reply:
Cool, makes sense.

Re side note: yes, agreed. There are workarounds: e.g. using a `try/catch` with `mutateAsync` or  `await new Promise(resolve => mutation.mutate(input, { onSuccess: resolve })` but it's not ideal. Sadly there's just no way to have unhandled promise rejections in React event handlers work nicely.